### PR TITLE
Feature/gui component settings

### LINKED ISSUE #238: Unify component settings

### PR DESCRIPTION
**Pull request by FelixOpolka**
closes #238 
Unifies component settings gui. If a component (e.g. Memory, Help, Registers, Output etc.) provides component settings, a button is displayed inside the component header.

### CODE REVIEW

File: source/ui/Components/Output/LightstripSettingsWindow.qml [hunk 1]
```diff
@@ -144,8 +144,8 @@ Window {
         anchors.bottomMargin: 5
 
         onClicked: {
-            baseAddressTextField.processInput();
-            numberOfStripsTextField.processInput();
+            baseAddressTextField.focus = false;
+            numberOfStripsTextField.focus = false;
```
Comment (riedersa) [inline source/ui/Components/Output/LightstripSettingsWindow.qml:148]: nice idea

**Review by riedersa — APPROVED** [review]
The changes worked on my Computer very well and i did not find any Problem in code

File: source/ui/BlankComponent.qml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import QtQuick 2.0
+import QtQuick.Controls 1.4
+
+Rectangle {
```
Comment (goldsborough) [inline source/ui/BlankComponent.qml]: Why?

File: source/ui/SplitviewItem.qml [hunk 3]
```diff
@@ -58,77 +59,113 @@ Item {
...
-            }
-            onHoveredChanged: {
-                if (hovered) {
-                    isExpanded=true
+            Item {
+                id: headerItems
+                anchors.fill: parent
+                visible: false
+
+                /*Choose*/
```
Comment (goldsborough) [inline source/ui/SplitviewItem.qml]: For the future, you can use `//` for single-line comments

**Review by goldsborough — APPROVED** [review]

**Pull request merged**